### PR TITLE
ADCM-3025 Test that audit objects don't duplicate or falsely changed on UPDATE

### PR DESCRIPTION
**Pull request by Sealwing**

### CODE REVIEW

**Review by DanBalalan — APPROVED** [review]

File: python/audit/tests/test_audit_objects.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+from rest_framework.status import HTTP_201_CREATED, HTTP_204_NO_CONTENT
+
+from adcm.tests.base import BaseTestCase
+from audit.models import AuditObject, AuditObjectType
+from cm.models import Bundle, Prototype, PrototypeConfig
+
+
+class TestAuditObjects(BaseTestCase):
+
+    cluster_proto: Prototype
```
Comment (a-alferov) [inline python/audit/tests/test_audit_objects.py:26]: I think it's better to get rid of the class variables, and transfer them to a `setUp()` or a function `_prepare_prototypes()`

**Review by a-alferov — CHANGES_REQUESTED** [review]

File: python/audit/tests/test_audit_objects.py (hunk 1 above)
Comment (Sealwing) [reply python/audit/tests/test_audit_objects.py:26]: Moved to `setUp` with assignment of `None` value to help IDE in type hints

File: python/audit/tests/test_audit_objects.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+                type=type_,
+                name=name,
+                version="2.3",
+                bundle=provider_bundle if type_ in {"provider", "host"} else cluster_bundle,
+            )
+            PrototypeConfig.objects.create(name="param", type="integer", prototype=proto)
+            setattr(self, name, proto)
+        self.service_proto.parent = self.cluster_proto
+        self.service_proto.save()
+        self.component_proto.parent = self.cluster_proto
```
[the remark this answers is not in the document]
Comment (Sealwing) [reply python/audit/tests/test_audit_objects.py:53]: Thanks, fixed

[the remark this answers is not in the document]
Comment (a-alferov) [reply python/audit/tests/test_audit_objects.py]: I don't see fixes

[the remark this answers is not in the document]
Comment (Sealwing) [reply python/audit/tests/test_audit_objects.py]: Ah, I've missed one with None as service parent,

**Review by a-alferov — APPROVED** [review]

**Pull request merged**